test-crawler/collector/collector_test.go

```diff
@@ -0,0 +1,46 @@
+package collector
+
+import (
+	"fmt"
+	"io/ioutil"
+	"os"
+	"strings"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+)
+
+func TestListGoFilesInFolder(t *testing.T) {
+	t.Run("Read golang files from temporary directory", func(t *testing.T) {
+		tempDir := t.TempDir()
+		err := ioutil.WriteFile(fmt.Sprintf("%s/a.go", tempDir), []byte("package main func main(){a:=1}"), 0644)
+		assert.NoError(t, err)
+		err = ioutil.WriteFile(fmt.Sprintf("%s/b.go", tempDir), []byte("package main func main(){b:=2}"), 0644)
+		assert.NoError(t, err)
+
+		system, files, err := ListGoFilesInFolder(tempDir, nil)
+		assert.NoError(t, err)
+		splitedFolderName := strings.Split(tempDir, "/")
+		systemName := splitedFolderName[len(splitedFolderName)-1]
+		assert.Equal(t, system, systemName)
+		assert.Equal(t, 2, len(files))
+		os.Remove(tempDir)
```

Comment:
Unnecessary: `t.TempDir()` should delete this directory. Please double-check.

Reply:
Yeah I also thought the same, but in the docs of ioutil package, it is sad:
`It is the caller's responsibility to remove the directory when no longer needed`

Reply:
That is a completely different package. Compare the `ioutil` package:
![image](https://user-images.githubusercontent.com/52637245/204330069-e5048a81-ef28-4b4b-affd-9c049552b24d.png)

With the `testing` package:
![image](https://user-images.githubusercontent.com/52637245/204330178-7690cf86-b2e4-42bb-a20b-1e8d8d3bee6d.png)

Reply:
Yeah that's correct. package testing is used for folder creation, and ioutil for writing into files. I will remove it